app/routes/joblistings/components/JoblistingEditor.js

```diff
@@ -24,195 +27,249 @@ type Props = {
-      ...newJoblisting,
-      workplaces
+    return this.props
+      .submitJoblisting({
+        ...newJoblisting,
+        id: this.props.joblistingId,
+        applicationUrl: httpCheck(newJoblisting.applicationUrl),
+        workplaces
+      })
+      .then(result => {
+        const id = this.props.joblistingId || result.payload.result;
+        this.props.push(`/joblistings/${id}/`);
+      })
+      .catch(err => {
+        if (err.payload && err.payload.response) {
+          throw new SubmissionError(err.payload.response.jsonData);
+        }
+      });
+  };
+
+  onDeleteJoblisting = () => {
+    this.props.deleteJoblisting(this.props.joblisting.id).then(() => {
+      this.props.push('/joblistings/');
     });
   };
 
-  if (!isNew && fetching) {
-    return <LoadingIndicator loading />;
+  fetchContacts = (company: SelectInputObject) => {
+    return this.props
+      .fetchCompanyContacts({ companyId: company.value })
+      .then(action => {
+        const responsibleOptions = action.payload.map(contact => ({
+          label: contact.name,
+          value: contact.id
+        }));
+        this.setState({ responsibleOptions });
+      });
+  };
+
+  componentDidMount() {
+    if (!this.props.isNew) {
+      this.fetchContacts(this.props.company);
+    }
   }
 
-  return (
-    <Content>
-      <h1 className={styles.heading}>
-        {!isNew ? 'Rediger jobbannonse' : 'Legg til jobbannonse'}
-      </h1>
-      <Form onSubmit={handleSubmit(onSubmit)}>
-        <Field
-          placeholder="Tittel på jobbannonse"
-          label="Tittel"
-          name="title"
-          component={TextInput.Field}
-          required
-        />
-        <Field
-          placeholder="Bedriften annonsen gjelder"
-          label="Bedrift"
-          name="company"
-          component={SelectInput.AutocompleteField}
-          filter={['companies.company']}
-          onChange={() =>
-            dispatch(
-              change('joblistingEditor', 'responsible', {
-                label: 'Ingen',
-                value: null
-              })
+  render() {
+    const { handleSubmit, isNew, dispatch, fetching = false } = this.props;
+
+    if (!isNew && fetching) {
+      return <LoadingIndicator loading />;
+    }
+
+    return (
+      <Content>
+        <h1 className={styles.heading}>
+          {!isNew ? 'Rediger jobbannonse' : 'Legg til jobbannonse'}
+        </h1>
+        <Form onSubmit={handleSubmit(this.onSubmit)}>
+          <Field
+            placeholder="Title"
+            label="Title"
+            name="title"
+            component={TextInput.Field}
+            required
+          />
+          <Field
+            placeholder="Bedrift"
+            label="Bedrift"
+            name="company"
+            component={SelectInput.AutocompleteField}
+            filter={['companies.company']}
+            onChange={company => {
+              this.fetchContacts(company).then(() => {
+                dispatch(
```

Comment:
this seems a bit icky, what's going on here?

Reply:
Not exactly sure, as @tmsolber wrote this..

My initial understanding: When chainging company, the companyContact field is cleared, and the companyContacts for the newly selected company are fetched...